src/goto-symex/goto_symex_state.cpp

```diff
@@ -132,13 +132,17 @@ static bool check_renaming(const exprt &expr)
   if(check_renaming(expr.type()))
     return true;
 
-  if(expr.id()==ID_address_of &&
-     expr.op0().id()==ID_symbol)
-    return check_renaming_l1(expr.op0());
-  else if(expr.id()==ID_address_of &&
-          expr.op0().id()==ID_index)
-    return check_renaming_l1(expr.op0().op0()) ||
-           check_renaming(expr.op0().op1());
+  if(
+    expr.id() == ID_address_of &&
+    to_address_of_expr(expr).object().id() == ID_symbol)
+    return check_renaming_l1(to_address_of_expr(expr).object());
+  else if(
+    expr.id() == ID_address_of &&
+    to_address_of_expr(expr).object().id() == ID_index)
+    return check_renaming_l1(
+             to_index_expr(to_address_of_expr(expr).object()).array()) ||
+           check_renaming(
+             to_index_expr(to_address_of_expr(expr).object()).index());
```

Comment:
Actually this code should be restructured to do the `to_address_of_expr` just once as it isn't zero cost.

Reply:
ok